_posts/2018-10-23-artsy-auth.md

```diff
@@ -0,0 +1,269 @@
+   the user that gravity shared.
+
+There's a slight difference in authenticating a _user_ and authenticating a _trusted application_. If our incoming
+request was generated by another Artsy service then we don't redirect anyone to gravity for login - rather, we
+manually create the token in advance, and store it. Once this is set up, the above process exits after step 2. But
+for now, let's continue looking at how user authentication works, because it's important to understand the whole
+system. We'll circle back to app authentication at the end.
+
+So!
+
+Think about it this way: when the user is redirected to gravity for login, gravity gathers up a bunch of
+information about who this user is. What's their user ID? What roles has their account been assigned? It lumps
+these together to create a JWT, which is a short statement of identity verification. It sticks this JWT into the
+user's headers - conceptually, think of this as a legal document or a letter of introduction - "The bearer of this
+document is User 123. Sincerely, Gravity."
+
+In fact, with OAuth the full header looks like this: `Authorization: Bearer <Encrypted JWT>`
+
+So, what's in that JWT? That's a great question! First, let's talk briefly about how encryption works just to make
+sure we're all on the same page. Then we'll dig into our JWTs and how to use them to provision access to parts of
+our applications.
+
+#### Shared Secret
+
+This document is not a guide to encryption, but a basic understanding of private key encryption is helpful to
+understand how this stuff works. Let's say you have some message `MSG` and you want to encrypt it. You can use some
+random secret key, let's call it `ABC`, to encrypt your message. There are a number of ways to do this, let's not
+worry about the implementation - assume that you have access to a function called `encrypt(message, secret)`. So if
+we want to encrypt our message, we can say `encrypt(MSG, ABC)` and we'll get back a string of meaningless
+characters, let's call it `ENC`. The key here is that there also exists a function `decrypt(encrypted, secret)`,
+such that as long as the secret is the same will convert your encrypted string back into its original state.
+`decrypt(ENC, ABC) === MSG`
+
+This is basically how security works on much of the internet, with various twists that optimize for different
+things depending on what exactly is being secured and from whom. If you want to send a message out into the world
+but you only want a specific person to be able to read it, you can encrypt it with your secret and then share the
+secret with only those people that you want to allow to read your message.
+
+This is what's going on with our `Authorization` token. Gravity encrypts user data using a secret key, and under
+the hood you've got that same secret key in your application. This allows your app to decrypt the value of the
+`Authorization` token and get at the JWT that it contains. So what's _that_ look like?
```

Comment:
This whole section is a really great explanation of the _payload_ of JWTs, but it makes it sound like JWTs themselves are encrypted (instead of having their header and payload cryptographically signed). It's a small distinction, but I think it's important to note that data we put into the `payload` is readable in plaintext.